### DISCUSSION TITLE
CIP Safety

### DISCUSSION THREAD

**Discussion by philrittenhouse** [ottowayi/pycomm3#200] [Q&A] [answered]
Hi,

I did a quick scan of the docs but wasn't able to find any details on CIP Safety support.   Does anyone know if this is supported or if something exists that builds on pycomm3 to support it?

Best regards,
Phil

**Comment by ottowayi** [comment, marked as answer]
There's nothing in there about CIP Safety mostly just because I don't have the spec for it and no one has requested it before.  You may be able to use generic messaging to build your own requests, but I have never worked with it before so I can't say for sure if it will work or not.  If CIP Safety objects can use explicit messaging (not an I/O connection) or if you have an example of something using a Logix MSG then it should work. 
The library is focused around accessing tags in a PLC, but I'm currently working on adding more general CIP functionality.  Right now the `generic_message` method is really the only way to use unsupported features, but it can take some trial and error unless you're pretty familiar with protocol.  I'm working on improving that as well.  Sorry I don't have a solid answer for you, but if there is a specific object or something you want to read I can try and help you put together a generic message.
